deploy/osd-machine-api/010-machine-api.MachineHealthCheck.yaml

```diff
@@ -0,0 +1,18 @@
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineHealthCheck
+metadata:
+  name: srep-healthcheck
+  namespace: openshift-machine-api
+spec:
+  selector:
+    matchLabels:
+      machine.openshift.io/cluster-api-machine-role: worker
+      machine.openshift.io/cluster-api-machine-role: infra
+  unhealthyConditions:
+  - type:    "Ready"
+    timeout: "480s"
+    status: "False"
+  - type:    "Ready"
+    timeout: "480s"
+    status: "Unknown"
```

Comment:
I want to call this out as the actual healthcheck going in.

If a worker or infra node goes unready or unavailable for 480s (8m) then it fails the healthcheck and the machine-api(-operator?) will replace it.

I chose 8 minutes because I know a newly provisioned node usually takes ~6 minutes to join the cluster. Obviously this is a number that we can tune.